api/src/Entity/Period.php

```diff
@@ -232,6 +232,23 @@ public function removeScheduleEntry(ScheduleEntry $scheduleEntry): self {
         return $this;
     }
 
+    /**
+     * All the content nodes used in some activity which is carried out (has a schedule entry) in this period.
+     *
+     * @return ContentNode[]
+     */
+    #[ApiProperty(writable: false, example: '["/content_nodes/1a2b3c4d"]')]
+    #[RelatedCollectionLink(ContentNode::class, ['period' => '$this'])]
+    #[Groups(['read'])]
+    public function getContentNodes(): array {
```

Comment:
I think we should check the performance impact of this. Ideally after merging https://github.com/ecamp/ecamp3/pull/2683 and then verifying period endpoint before and after.

With our current implementation, this means all content nodes are always loaded whenever a period entity is normalized (e.g. when loading or patching a single period, when loading all periods, when loading a camp with embedded periods, etc.)

Reply:
Yeah, good point. Actually, for HAL the result of this calculation is ignored because we render a RelatedCollectionLink, and we could just `return [];`. But I implemented it for formats other than HAL, and we should define a way to cope with this.

The same problem happens in Day#getScheduleEntries.

Reply:
Status:
* I opened issue #2829 to continue the discussion
* Right now react pdf is broken anyways, I am working on a fix over at the react-pdf codebase. Once that is resolved, we can continue looking at this PR.